File: algorithms/easy/longest_subsequence_with_limited_sum.cpp
class Solution {
public:
    vector<int> answerQueries(vector<int>& nums, vector<int>& queries) {
      sort(nums.begin(), nums.end());
      int sum = 0;
      for (int i = 0; i < nums.size(); i++) {
        sum += nums[i];
        nums[i] = sum;
      }
      vector<int> ans;
      for (auto q : queries) {
        auto n = upper_bound(nums.begin(), nums.end(), q);
        ans.push_back(distance(nums.begin(), n));
      }
      return ans;
    }
};

File: algorithms/easy/longest_subsequence_with_limited_sum.rs
impl Solution {
    pub fn answer_queries(mut nums: Vec<i32>, queries: Vec<i32>) -> Vec<i32> {
        nums.sort();
        let mut sum = 0;
        for i in 0..nums.len() {
            sum += nums[i];
            nums[i] = sum;
        }

        let mut ans = Vec::new();
        for query in queries {
            let i = nums.partition_point(|&x| x <= query);
            ans.push(i as i32 + 1);
        }
        ans
    }
}
